### ISSUE TITLE
Date foreign key references produce duplicates in filter and group by dialogs

### ISSUE THREAD

**Issue by notrom** [metabase/metabase#15563]
**Describe the bug**
Joins defined in the data model using a datetime as the key are giving duplicated rows in filter and group by menus.

We have date fields in our data that we use as a foreign key to a date table, primary key is a date. We model this in the Metabase data model. Previously this has work well for us but I've just upgraded a test instance to v0.39.0-rc1 and now we're seeing some strange behaviour in the UI with multiple duplicate fields being displayed.

The resulting summary is correct for each month and the generated query seems to be correct as I'd expect.

**Logs**
Nothing of interest

**To Reproduce**
1. In Metabase data model, Sample dataset, change `Orders > Created At` to be the entity key
2. Change `Reviews > Created At` to be a foreign key referencing the 'Orders > Created At'
3. Make a new question for Reviews
4. Summarize, now note that "Created At" is repeated multiple times
5. Filter, same thing repeated.

**Expected behavior**
Dates should be able to be used as primary and foreign keys without causing UI issues.

**Screenshots**
These are from our data, not Sample Dataset.

In v0.38.1 the group by options for these fields looks like this:
![image](https://user-images.githubusercontent.com/4504437/114336314-c3fbd300-9ba2-11eb-9b38-9cb7dd9e8170.png)

Now in v0.39.0-rc1 it looks like this:
![image](https://user-images.githubusercontent.com/4504437/114336342-d2e28580-9ba2-11eb-8092-67ee3882bbca.png)

When selecting any one of these repeats they all get highlighted
![image](https://user-images.githubusercontent.com/4504437/114336427-0de4b900-9ba3-11eb-8a97-aa28c4a66073.png)

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-NZ",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.90 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "Cp1252",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "13.0.2+8",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "https://java.oracle.com/",
    "java.version": "13.0.2",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "13.0.2+8",
    "os.name": "Windows 10",
    "os.version": "10.0",
    "user.language": "en",
    "user.timezone": "Pacific/Auckland"
  },
  "metabase-info": {
    "databases": [
      "sqlserver",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.1"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.18"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-04-06",
      "tag": "v0.39.0-rc1",
      "branch": "release-x.39.x",
      "hash": "471eda1"
    },
    "settings": {
      "report-timezone": "UTC"
    }
  }
}
```

**Severity**
Makes the grouping and filtering dialogs harder to use, more importantly it just looks wrong. Functionality still works fine.

**Additional context**
Other relationships seems to be fine, it's just dates that are causing this.

**Comment by flamber** [comment]
Related to #15446 and #15574, since the duplicated fields are actually belonging to Reviews
![image](https://user-images.githubusercontent.com/1447303/114446613-bc075780-9bd1-11eb-9ac9-d55280394658.png)

**Comment by ariya** [comment]
Looks like this is also fixed by PR #15582 (already merged to release-x.39.x branch).

**Comment by flamber** [comment]
@ariya :+1: fixed by #15582
![image](https://user-images.githubusercontent.com/1447303/114626614-43c59280-9cb4-11eb-8364-77ba788761cd.png)
